core/trino-main/src/main/java/io/trino/operator/PipelineContext.java

```diff
@@ -145,17 +149,19 @@ public boolean isOutputPipeline()
 
     public DriverContext addDriverContext()
     {
-        return addDriverContext(Lifespan.taskWide());
+        return addDriverContext(null, Lifespan.taskWide());
     }
 
-    public DriverContext addDriverContext(Lifespan lifespan)
+    public DriverContext addDriverContext(@Nullable ScheduledSplit partitionedSplit, Lifespan lifespan)
```

Comment:
I went with just `long splitWeight` instead since any other alternative was a needless short term object allocation and it made more sense to me to just add some extra validation checks.